### PR TITLE
[DDW-745] Daedalus aarch64 (a.k.a M1) native support

### PR DESCRIPTION
**Pull request by danielmain**
This PR adds ARM native support for Daedalus.

Before officially giving those builds to the users, we need to wait for devops-approved `aarch64-darwin` hardware. For now we’re using Buildkite workers provided by @angerman – thank you! :bow: :bow: :bow: 

## Todos

*n/a*

## Screenshots

*n/a*

## Testing Checklist

- [Slack QA thread](https://input-output-rnd.slack.com/archives/GGKFXSKC6/p1653056529530429)
- [Regression document](https://docs.google.com/spreadsheets/d/1WgLlRPtppJTNiOkBG0T_GhGJmJ552h-uCXLP153ATjA/edit?usp=sharing)

---

### Basics

- [x] PR assigned to the PR author(s)
- [x] `input-output-hk/daedalus-dev` and `input-output-hk/daedalus-qa` assigned as PR reviewers
- [ ] If there are UI changes, Alexander Rukin assigned as an additional reviewer
- [ ] All visual regression testing has been reviewed (assign `run Chromatic` label to PR to trigger the run)
- [x] PR has appropriate labels (`release-vNext`, `feature`/`bug`/`chore`, `WIP`)
- [x] PR link is added to a Jira ticket, ticket moved to In Review
- [x] PR is updated to the most recent version of the target branch (and there are no conflicts)
- [x] PR has a good description that summarizes all changes
- [ ] PR contains screenshots (in case of UI changes)
- [x] CHANGELOG entry has been added to the top of the appropriate section (_Features_, _Fixes_, _Chores_) and is linked to the correct PR on GitHub
- [ ] There are no missing translations (running `yarn manage:translations` produces no changes)
- [ ] Text changes are proofread and approved (Jane Wild / Amy Reeve)
- [ ] Japanese text changes are proofread and approved (Junko Oda)
- [ ] Storybook works and no stories are broken (`yarn storybook`)
- [x] In case of dependency changes `yarn.lock` file is updated

### Code Quality

- [x] Important parts of the code are properly commented and documented
- [x] Code is properly typed with typescript types
- [x] React components are split-up enough to avoid unnecessary re-renderings
- [x] Any code that only works in main process is neatly separated from components

### Testing

- [ ] New feature/change is covered by acceptance tests
- [ ] New feature/change is manually tested and approved by QA team
- [ ] All existing acceptance tests are still up-to-date
- [ ] New feature/change is covered by Daedalus Testing scenario
- [ ] All existing Daedalus Testing scenarios are still up-to-date

### After Review

- [ ] Update Slack QA thread by marking it with a green checkmark

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -14,6 +14,7 @@
 
 ### Features
 
+- Support of apple aarch64 chip ([PR ](https://github.com/input-output-hk/daedalus/pull/))
```
Comment (mark-stopka) [inline CHANGELOG.md:17]: Pull request number is missing...

File: shell.nix [hunk 2]
```diff
@@ -57,12 +57,14 @@ let
       git python27 curl jq
       nodePackages.node-gyp nodePackages.node-pre-gyp
       gnumake
-      chromedriver
+      (if system == "aarch64-darwin"
+        then (localLib.iohkNix.getPkgs { system = "x86_64-darwin"; inherit config; }).chromedriver
```
Comment (Diatrus) [inline shell.nix:61]: In https://github.com/NixOS/nixpkgs/commit/6e0a2c1c5422706447a34f71c48f6c37f0a6542d#diff-e846459ce0553983db1a53479d09dc25fe71fab9f13db893ab85f35d15b6172d nix added support for aarch64 chromedriver, removing the need for this

Comment (danielmain) [reply shell.nix:61]: Thank you @Diatrus!

**Comment by michalrus** [comment]
Current status update: @danielmain is able to get inside `nix-shell` on M1 now.

**Comment by michalrus** [comment]
Current status update: once we get Buildkite agents for Apple M1, it should work at the current commit – ARM installers tested locally by @danielmain and me.

File: installers/nix/electron.nix [hunk 3]
```diff
@@ -12,7 +12,7 @@ let
     homepage = https://github.com/electron/electron;
     license = licenses.mit;
     maintainers = with maintainers; [ travisbhartwell manveru ];
-    platforms = [ "x86_64-darwin" "x86_64-linux" "i686-linux" "armv7l-linux" "aarch64-linux" ];
+    platforms = [ "x86_64-darwin" "aarch64-darwin" "x86_64-linux" "i686-linux" ];
```
Comment (michalrus) [inline installers/nix/electron.nix:15]: Not sure why `i686` was added?

Comment (michalrus) [inline installers/nix/electron.nix:15]: Why do we have `i686-linux` here?

File: installers/dhall-json.nix [hunk 4]
```diff
@@ -1,6 +1,6 @@
 { mkDerivation, aeson, aeson-pretty, base, bytestring, dhall
-, fetchgit, optparse-generic, stdenv, text, trifecta, lib
-, unordered-containers, yaml
+, fetchgit, optparse-generic, stdenv, text, trifecta
+, unordered-containers, yaml, lib
```
Comment (michalrus) [inline installers/dhall-json.nix:3]: diff noise :thinking:

File: installers/dhall-haskell.nix [hunk 5]
```diff
@@ -6,7 +6,7 @@
 , memory, mtl, optparse-generic, parsers, prettyprinter
 , prettyprinter-ansi-terminal, repline, scientific, stdenv, tasty, lib
 , tasty-hunit, text, text-format, transformers, trifecta, unordered-containers
-, vector
+, vector, lib
```
Comment (michalrus) [inline installers/dhall-haskell.nix:9]: unused

File: installers/daedalus-installer.nix [hunk 6]
```diff
@@ -1,9 +1,9 @@
 { mkDerivation, aeson, base, bytestring, containers, directory
 , filepath, foldl, github, Glob, hspec, lens-aeson, managed
 , megaparsec, microlens, network-uri, nsis, optional-args
-, optparse-applicative, optparse-generic, split, lib
+, optparse-applicative, optparse-generic, split, stdenv
```
Comment (michalrus) [inline installers/daedalus-installer.nix:4]: unused, noise

File: nix/launcher-config.nix [hunk 7]
```diff
@@ -34,6 +34,7 @@ let
   configDir = configFilesSource: {
     linux = configFilesSource;
     macos64 = if devShell then configFilesSource else "\${DAEDALUS_INSTALL_DIRECTORY}/../Resources";
+    macos64-arm = if devShell then configFilesSource else "\${DAEDALUS_INSTALL_DIRECTORY}/../Resources";
```
Comment (michalrus) [inline nix/launcher-config.nix:37]: I would very much like to turn all those instances in our source code to standard arch+os names like `aarch64-darwin`. Why complicate?

File: nix/launcher-config.nix [hunk 8]
```diff
@@ -105,20 +108,23 @@ let
   dataDir = let
     path.linux = "\${XDG_DATA_HOME}/Daedalus/${network}";
     path.macos64 = "\${HOME}/Library/Application Support/${spacedName}";
+    path.macos64-arm = "\${HOME}/Library/Application Support/${spacedName}";
```
Comment (michalrus) [inline nix/launcher-config.nix:111]: it could just as well say `path.macos64-arm = path.macos64`, DRY-ier

File: nix/sources.json [hunk 9]
```diff
@@ -12,27 +12,27 @@
...
+        "rev": "0d1d5f036c73d18e641412d2c58d4acda592d493",
+        "sha256": "05cqxzrcvsqrmv22clf9l5agwzrcrcjw33mpmc4fnim8np5ym6w3",
         "type": "tarball",
-        "url": "https://github.com/input-output-hk/cardano-shell/archive/5257fc818ef7df60986834310432e197aac90711.tar.gz",
+        "url": "https://github.com/input-output-hk/cardano-shell/archive/0d1d5f036c73d18e641412d2c58d4acda592d493.tar.gz",
         "url_template": "https://github.com/<owner>/<repo>/archive/<rev>.tar.gz"
     },
     "cardano-wallet": {
-        "branch": "v2022-04-27",
+        "branch": "hkm/enable-aarch64-darwin",
```
Comment (michalrus) [inline nix/sources.json:27]: wasn’t this merged?

File: scripts/build-installer-unix.sh--nix-2.5 [hunk 10]
```diff
@@ -0,0 +1,19 @@
+#!/usr/bin/env bash
+set -e
+
+echo '~~~ Obtaining pkgs.nixUnstable'
```
Comment (michalrus) [inline scripts/build-installer-unix.sh--nix-2.5:4]: Can we get devops to update Nix on Buildkite CI workers…?

File: shell.nix [hunk 11]
```diff
@@ -113,11 +115,16 @@ let
...
       ln -svf $(type -P cardano-cli)
       mkdir -p ${BUILDTYPE}/
-      ln -svf $PWD/node_modules/usb/build/${BUILDTYPE}/usb_bindings.node ${BUILDTYPE}/
-      ln -svf $PWD/node_modules/node-hid/build/${BUILDTYPE}/HID.node ${BUILDTYPE}/
-      ln -svf $PWD/node_modules/node-hid/build/${BUILDTYPE}/HID_hidraw.node ${BUILDTYPE}/
-      # XXX: right now we only build Release/detection.node (TODO: investigate why – @michalrus)
-      ln -svf $PWD/node_modules/usb-detection/build/Release/detection.node ${BUILDTYPE}/
+      ${let
+        # XXX: right now we don’t build Debug/ versions on Linux (TODO: investigate why – @michalrus)
+        sourceBUILDTYPE = if system == "x86_64-linux" then "Release" else BUILDTYPE;
```
Comment (michalrus) [inline shell.nix:120]: I don’t understand this

File: yarn2nix.nix [hunk 12]
```diff
@@ -212,11 +203,15 @@ yarn2nix.mkYarnPackage {
   '';
   pkgConfig = {
     electron-rebuild = {
+      # TODO: That is rather awful… Can it be done better? – @michalrus
```
Comment (michalrus) [inline yarn2nix.nix:206]: it is rather awful…

**Review by michalrus — CHANGES_REQUESTED** [review]

**Review by michalrus — APPROVED** [review]

**Review by dmitrii-gaico — APPROVED** [review]

**Review by miorsufianiohk — APPROVED** [review]
LGTM. Great work @michalrus 👍 . Tested on 21812

**Review by michalrus — APPROVED** [review]

**Pull request merged**